fw/http_frame.c

```diff
@@ -1951,7 +1994,37 @@ tfw_h2_frame_process(TfwConn *c, struct sk_buff *skb, struct sk_buff **next)
 			kfree_skb(nskb);
 			goto out;
 		}
-	} else {
+	}
+	else if (unlikely(tfw_h2_allowed_empty_frame(h2))) {
+		/*
+		 * Process empty frames.
+		 */
+		struct sk_buff *pskb, *end = h2->skb_head->prev;
+
+		/*
+		 * Free all SKBs in queue except the last. The last one
+		 * will be passed to message processing function.
+		 */
+		while (unlikely(h2->skb_head != end)) {
```

Comment:
I am not sure here. I think that the loop from previous path of code should be here. Can we have some skbs which are not relate to this stream here? Is here a chance that we drop them?

Reply:
No, if we have skbs not related to current stream during processing frame it's protocol violation.